service.js

```diff
@@ -84,12 +85,20 @@ module.exports = ({bus, logFactory}) => {
             });
         }
 
+        if (Array.isArray(serviceConfig.errors)) {
+            serviceConfig.errors.forEach(errorFactory => {
+                if (errorFactory instanceof Function) {
+                    errorFactory(bus.errors);
+                }
+            });
+        }
+
         return servicePorts.create(ports, config, test);
     };
 
     let create = (serviceConfig, config, test) => {
         if (typeof serviceConfig === 'function') {
-            return new Promise(resolve => resolve(serviceConfig({config, bus: bus.publicApi})))
+            return Promise.resolve(serviceConfig({config, bus: bus.publicApi}))
```

Comment:
This may throw, while the old code will return a rejected promise.
Is this the idea of the change?

Reply:
No. Will revert